coordinator/internal/seedengine/seedengine.go

```diff
@@ -82,15 +79,12 @@ func New(secretSeed []byte, salt []byte) (*SeedEngine, error) {
 	return se, nil
 }
 
-// DerivePodSecret derives a secret for a pod from the policy hash and the secret seed.
-func (s *SeedEngine) DerivePodSecret(policyHash [hashSize]byte) ([]byte, error) {
-	if policyHash == [hashSize]byte{} {
-		return nil, errors.New("policy hash must not be empty")
-	}
-	if bytes.Equal(policyHash[:], s.hashFun().Sum(nil)) {
-		return nil, errors.New("policy hash is the hash of an empty byte slice")
+// DeriveWorkloadSecret derives a secret for a workload from the workload name and the secret seed.
+func (s *SeedEngine) DeriveWorkloadSecret(workloadSecretID string) ([]byte, error) {
```

Comment:
I think the name "DerivePodSecret" signals that the derived secret is pod specific. Also, since there wouldn't be any other consumers of `DerivePodSecret` I'd unexport it and also the parameter needs to be renamed. Then I think there's no reason to hash something that will be part of the info argument (I would think one of the reasons why the RFC specifies putting the string directly into the info argument).

Reply:
I see, thanks for clarifying! :)